### PR TITLE
Mtracy/status on feed

### PR DESCRIPTION
**Pull request by mrtracy**
Two commits:
- First commit removes unused file "status/status.go", and removes the unqualified "/_status/" which was unused and just returned an empty object.
- Second, larger commit starts publishing periodic status information to the store event feed. This brings the information available in "status/monitor.go" up to parity with the store status scanner. This will allow us to _remove_ the store status scanner in a subsequent commit.

### CODE REVIEW

File: server/status/monitor.go [hunk 1]
```diff
@@ -44,6 +50,7 @@ type NodeStatusMonitor struct {
 	sync.RWMutex
 	stores     map[proto.StoreID]*StoreStatusMonitor
 	nodeID     proto.NodeID
+	desc       proto.NodeDescriptor
```
Comment (tamird) [inline server/status/monitor.go]: `NodeDescriptor` has a `NodeID` on it. Remove `nodeID` on the preceding line?

File: storage/feed.go [hunk 2]
```diff
@@ -79,6 +79,34 @@ type StartStoreEvent struct {
 	StoreID proto.StoreID
 }
 
+// StoreStatusEvent contains the current descriptor for the given store.
+//
+// Because the descriptor contains information that cannot currently be computed
+// from other events, this event should be periodically broadcast by the store
+// independently of other operations.
+type StoreStatusEvent struct {
+	StoreID proto.StoreID
```
Comment (tamird) [inline storage/feed.go]: StoreDescriptor has a StoreID on it

File: storage/feed.go [hunk 3]
```diff
@@ -79,6 +79,34 @@ type StartStoreEvent struct {
...
+	Desc    *proto.StoreDescriptor
```
Comment (tamird) [inline storage/feed.go]: shouldn't this hang off the `StoreEventFeed`?

File: storage/store.go [hunk 4]
```diff
@@ -1649,6 +1665,25 @@ func (s *Store) updateStoreStatus() {
...
 }
 
+// PublishStatus publishes periodically computed status events to the store's
+// events feed. This method itself should be periodically called by some
+// external mechanism.
+func (s *Store) PublishStatus() {
+	// broadcast store descriptor.
+	desc, err := s.Descriptor()
+	if err != nil {
+		log.Error(err)
```
Comment (tamird) [inline storage/store.go]: why not return this error?

File: storage/feed.go (hunk 2 above)
Comment (mrtracy) [inline storage/feed.go]: Seemed better to me to have the redundant information to keep consistency with the rest of the events.

File: storage/feed.go (hunk 3 above)
Comment (mrtracy) [inline storage/feed.go]: The Desc can actually change - for some reason, it includes a store Capacity, which is recalculated every time Store.Descriptor() is called.

File: storage/feed.go (hunk 2 above)
Comment (tamird) [inline storage/feed.go]: is there some common interface being implemented?

File: storage/feed.go (hunk 3 above)
Comment (tamird) [inline storage/feed.go]: what do you mean by "change"? is it being updated in place or replaced entirely?

if it's the former, all the pointers will point to the same descriptor anyway, so i don't see why that'd preclude you from putting the pointer on the `StoreEventFeed`.

File: storage/store.go (hunk 4 above)
Comment (mrtracy) [inline storage/store.go]: Fixed to return, now handled in the node.

File: server/status/monitor.go (hunk 1 above)
Comment (mrtracy) [inline server/status/monitor.go]: `desc` wasn't being used yet. I've removed `desc`, I'll swap it out with `nodeID` when I need it (likely in my next PR).

File: storage/feed.go (hunk 3 above)
Comment (mrtracy) [inline storage/feed.go]: It still wouldn't help, the status module doesn't see a `StoreEventFeed`, it just has a subscription to a generic feed.  `StoreEventFeed` is just a helper class for constructing and publishing store-specific events.

Comment (tamird) [inline storage/feed.go]: OK now i'm confused. Is `StoreStatusMonitor.desc` being used?

Comment (tamird) [inline storage/feed.go]: i see it being set, but never inspected.

Comment (mrtracy) [inline storage/feed.go]: It's not being used yet.  It will be used in an immediately subsequent PR.

File: storage/feed.go (hunk 2 above)
Comment (mrtracy) [inline storage/feed.go]: No. I went ahead and removed it, you convinced me.

File: storage/feed.go (hunk 3 above)
Comment (tamird) [inline storage/feed.go]: Can you please remove it for now, then?

**Comment by mrtracy** [comment]
Feedback addressed, PTAL

File: storage/feed.go [hunk 5]
```diff
@@ -79,6 +79,33 @@ type StartStoreEvent struct {
 	StoreID proto.StoreID
 }
 
+// StoreStatusEvent contains the current descriptor for the given store.
+//
+// Because the descriptor contains information that cannot currently be computed
+// from other events, this event should be periodically broadcast by the store
+// independently of other operations.
+type StoreStatusEvent struct {
+	Desc *proto.StoreDescriptor
```
Comment (tamird) [inline storage/feed.go]: this is still not being used, can you remove it? it seems much more natural to me to introduce this in the same commit that actually uses it.

Comment (mrtracy) [inline storage/feed.go]: I opted to simply use it in this commit: i'm now recording capacity statistics as time series.  This is actually something helpful that I should have done anyway.

File: storage/store.go [hunk 6]
```diff
@@ -1625,8 +1630,19 @@ func (s *Store) updateStoreStatus() {
 			}
 		}
 	}
-	s.mu.Unlock()
+	return
```
Comment (tamird) [inline storage/store.go]: i believe this `return` is superfluous

Comment (tschottdorf) [inline storage/store.go]: no, you need it if your function returns a value (named or not).

Comment (tamird) [inline storage/store.go]: right, but this function doesn't return anything.

Comment (tamird) [inline storage/store.go]: oh, disregard, line wrapping threw me off =/

**Comment by mrtracy** [comment]
Any chance of an LGTM on this?

**Comment by BramGruneir** [comment]
LGTM

**Comment by tamird** [comment]
LGTM

**Pull request merged**